server/container/testing/src/test/java/org/apache/james/deployment/DeploymentTest.java

```diff
@@ -0,0 +1,190 @@
+import org.awaitility.Duration;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.utility.MountableFile;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+import net.javacrumbs.jsonunit.assertj.JsonAssertions;
+
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+public class DeploymentTest {
+
+    private static final String LOCALHOST = "localhost";
+    private static final String SIMPSON = "simpson";
+    private static final String HOMER = "homer@" + SIMPSON;
+    private static final String HOMER_PASSWORD = "secret";
+    private static final String BART = "bart@" + SIMPSON;
+    private static final String BART_PASSWORD = "tellnobody";
+    private static final int JMAP_PORT = 80;
+    private static final int SMTP_PORT = 25;
+    private static final int IMAP_PORT = 143;
+    private static final int WEBADMIN_PORT = 8000;
+
+    private static final Logger logger = LoggerFactory.getLogger(DeploymentTest.class);
+
+    public Network network = Network.newNetwork();
+
+    public GenericContainer<?> cassandra = new CassandraContainer()
+        .withNetwork(network)
+        .withNetworkAliases("cassandra");
+    
+    public GenericContainer<?> elasticsearch = new GenericContainer<>("elasticsearch:2.4.6")
+        .withNetwork(network)
+        .withNetworkAliases("elasticsearch");
+
+    public GenericContainer<?> james = new GenericContainer<>("linagora/james-project:latest")
```

Comment:
We bqse ourselves on not yet published images... Can't we rather rely on env variables?

Reply:
yes, of course